lib/implore.es6.js

```diff
@@ -194,17 +194,16 @@ implore.delete = function httpDelete(options) {
 function getURLFromRequest(request) {
 	const queryString = makeQueryString(request.query || {});
 	let formatted = request.route;
-	let name;
 	let value;
 	let regexp;
 
-	for (name in request.params) {
-		if (request.params.hasOwnProperty(name)) {
-			value = request.params[name];
-			regexp = new RegExp(':' + name + '(?=(\\\/|$))');
-			formatted = formatted.replace(regexp, value);
-		}
-	}
+	// Replace route params with their values. If the route param is specified
+	// with an extension (e.g., ':id.json'), keep the extension.
+	Object.keys(request.params || {}).forEach((name) => {
```

Comment:
I think the fluxthis developers prefer their [each](https://github.com/addthis/fluxthis/blob/master/lib/each.es6.js) helper